File: sdk/nodejs/src/resources/partner.js
"use strict";

const { BaseResource } = require("./base");

class PartnerResource extends BaseResource {
  async getPartnerInfo() {
    return this._http.get("$/api/partner");
  }

  async apply(data) {
    return this._http.post("$/api/partner/apply", data);
  }

  async listTiers() {
    return this._http.get("$/api/partner/tiers");
  }

  async getPricing() {
    return this._http.get("$/api/partner/pricing");
  }

  async listCustomers() {
    return this._http.get("$/api/partner/customers");
  }

  async getEarnings() {
    return this._http.get("$/api/partner/earnings");
  }

  async getWallet() {
    return this._http.get("$/api/partner/wallet");
  }

  async listPayouts() {
    return this._http.get("$/api/partner/payouts");
  }

  async requestPayout(data) {
    return this._http.post("$/api/partner/payouts", data);
  }

  async listLeads() {
    return this._http.get("$/api/partner/leads");
  }

  async createLead(data) {
    return this._http.post("$/api/partner/leads", data);
  }

  async getReferral() {
    return this._http.get("$/api/partner/referral");
  }

  async getRates() {
    return this._http.get("$/api/partner/rates");
  }

}

module.exports = { PartnerResource };

File: sdk/csharp/Resources/Partner.cs
namespace HiTechCloud.SDK.Resources;

public class Partner : BaseResource
{
    public Partner(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> InfoAsync()
    {
        return await Http.GetAsync("/api/partner");
    }

    public async Task<Dictionary<string, object?>> ApplyAsync(Dictionary<string, object> data)
    {
        return await Http.PostAsync("/api/partner/apply", data);
    }

    public async Task<Dictionary<string, object?>> TiersAsync()
    {
        return await Http.GetAsync("/api/partner/tiers");
    }

    public async Task<Dictionary<string, object?>> PricingAsync()
    {
        return await Http.GetAsync("/api/partner/pricing");
    }

    public async Task<Dictionary<string, object?>> CustomersAsync()
    {
        return await Http.GetAsync("/api/partner/customers");
    }

    public async Task<Dictionary<string, object?>> EarningsAsync()
    {
        return await Http.GetAsync("/api/partner/earnings");
    }

    public async Task<Dictionary<string, object?>> WalletAsync()
    {
        return await Http.GetAsync("/api/partner/wallet");
    }

    public async Task<Dictionary<string, object?>> PayoutsAsync()
    {
        return await Http.GetAsync("/api/partner/payouts");
    }

    public async Task<Dictionary<string, object?>> RequestPayoutAsync(Dictionary<string, object> data)
    {
        return await Http.PostAsync("/api/partner/payouts", data);
    }

    public async Task<Dictionary<string, object?>> LeadsAsync()
    {
        return await Http.GetAsync("/api/partner/leads");
    }

    public async Task<Dictionary<string, object?>> CreateLeadAsync(Dictionary<string, object> data)
    {
        return await Http.PostAsync("/api/partner/leads", data);
    }

    public async Task<Dictionary<string, object?>> ReferralAsync()
    {
        return await Http.GetAsync("/api/partner/referral");
    }

    public async Task<Dictionary<string, object?>> RatesAsync()
    {
        return await Http.GetAsync("/api/partner/rates");
    }
}
